src/order/locally_finite.lean

```diff
@@ -0,0 +1,465 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.preimage
+
+/-!
+# Locally finite orders
+
+This file defines locally finite orders.
+
+A locally finite order is an order for which all bounded intervals are finite. This allows to make
+sense of `Icc`/`Ico`/`Ioc`/`Ioo` as lists, multisets, or finsets.
+Further, if the order is bounded above (resp. below), then we can also make sense of the
+"unbounded" intervals `Ici`/`Ioi` (resp. `Iic`/`Iio`).
```

Comment:
Could you please mention some important examples (at least `ℕ` and `ℤ`, but I guess also `fin n`). Are there other interesting examples? Maybe `X →₀ R` if `R` is a locally finite order?

Reply:
Well spotted! I hadn't thought about that one.